vault/activity_log.go

```diff
@@ -2019,25 +2027,20 @@ func (a *ActivityLog) loadConfigOrDefault(ctx context.Context, isReportingEnable
 	if err := configRaw.DecodeJSON(&config); err != nil {
 		return config, err
 	}
-
-	// check if the retention time is lesser than the default when reporting is enabled
-	if (config.RetentionMonths < ActivityLogMinimumRetentionMonths) && isReportingEnabled {
-		updatedConfig, err := a.setDefaultRetentionMonthsInConfig(ctx, config)
-		if err != nil {
-			return config, err
-		}
-		return updatedConfig, nil
-	}
 	return config, nil
 }
 
 // setDefaultRetentionMonthsInConfig sets the retention months in activity config with default value.
 // This supports upgrades from versions prior to set the new default ActivityLogMinimumRetentionMonths.
 func (a *ActivityLog) setDefaultRetentionMonthsInConfig(ctx context.Context, inputConfig activityConfig) (activityConfig, error) {
+	if a.core.perfStandby {
+		return inputConfig, nil
```

Comment:
so if this code is running on a perf standby, and there's an existing config in storage with a lower retention month value, is the idea that we return `inputConfig` from this function with the value that's lower than the default, and wait for invalidation from the active node to happen in order to update the value to the new default?

Reply:
yes, that's right. The right value should get replicated from the active node.

Reply:
That was my understanding. It looks like `invalidateConfigPerfStandby` calls `loadConfigOrDefault` first, which would pick up the changes; however, it looks like we're not modifying `a.RetentionMonths` in `SetConfigStandby`... that's a really good catch. Maybe we want to add that logic here as well?

Reply:
Actually, I wonder if this really matters, since only the active node should ever be reporting 🤔... I think maybe it makes sense to make the change regardless, but curious what you think @akshya96 @miagilepner.

Reply:
That's true, I'm not sure if the perf standby actually uses the retention months for anything

Reply:
Even though reporting only happens from the active node, The perf standbys would should still show the right value when we do a read on sys/internal/counters/config

Reply:
Looks like read on sys/internal/counters/config from perf standbys returns the right value. It returns the value from loadConfigOrDefault

Reply:
Also looks like a.retentionMonths is used by the active in all places.